### PR TITLE
Add various missing undocumented functions

### PR DESCRIPTION
**Pull request by srittau**
Mostly addresses #2149

### CODE REVIEW

File: stdlib/2and3/sre_compile.pyi [hunk 1]
```diff
@@ -5,6 +5,20 @@ import sys
 from sre_parse import SubPattern
 from typing import Any, List, Pattern, Tuple, Type, Union
 
+# SRE_* constants are undocumented
```
Comment (hauntsaninja) [inline stdlib/2and3/sre_compile.pyi:8]: The implementation re-exports from sre_constants, maybe we should match

File: stdlib/3/urllib/request.pyi [hunk 2]
```diff
@@ -1,12 +1,11 @@
-# Stubs for urllib.request (Python 3.4)
-
 import os
 import ssl
 import sys
 from email.message import Message
 from http.client import HTTPMessage, HTTPResponse, _HTTPConnectionProtocol
 from http.cookiejar import CookieJar
 from typing import IO, Any, Callable, ClassVar, Dict, List, Mapping, NoReturn, Optional, Sequence, Tuple, TypeVar, Union, overload
+from urllib.error import HTTPError as HTTPError
```
Comment (hauntsaninja) [inline stdlib/3/urllib/request.pyi:8]: I wasn't sure I agreed with this, but I see previously mentioned my objection: https://github.com/python/typeshed/issues/2149#issuecomment-421140124

Comment (srittau) [reply stdlib/3/urllib/request.pyi:8]: I'll revert this and instead filed https://github.com/python/cpython/pull/22304.

File: stdlib/2and3/sre_compile.pyi (hunk 1 above)
Comment (srittau) [reply stdlib/2and3/sre_compile.pyi]: Good catch, changed it accordingly.

**Review by hauntsaninja — APPROVED** [review]

**Pull request merged**